Nb/documentation and autocasting pixels (#1345)
deepchecks/deepchecks: docs/source/user-guide/vision/data-classes/DetectionData.rst

The Object Detection Data Class
===============================

> The DetectionData is a :doc:`data class </user-guide/vision/data-classes/index>` designated for object detection tasks.

Reviewer: ```suggestion
The DetectionData is a :doc:`data class </user-guide/vision/data-classes/index>` designed for object detection tasks.
```

> It is a subclass of the :class:`~deepchecks.vision.VisionData` class and is used to load and preprocess data for object

Reviewer: ```suggestion
It is a subclass of the :class:`~deepchecks.vision.VisionData` class and is used to help deepchecks load and interact with object detection data using a well defined format.
```

---------------------
Deepchecks' checks use the :func:`~deepchecks.vision.DetectionData.batch_to_labels` function in order to get the labels in the correct format.
The accepted label format for is a a list of length N containing tensors of shape (B, 5), where N is the number
of samples within a batch, B is the number of bounding boxes in the sample and each bounding box is represented by 5 values:
``(class_id, x_min, y_min, w, h)``.

>     x_min and y_min are the coordinates (in pixels) of the **lower left corner** of the bounding box, w

Reviewer: ```suggestion
    x_min and y_min are the coordinates (in pixels) of the **top left corner** of the bounding box, w
```
I'm pretty sure it's top left...

Reviewer 2: It's top left

    and h are the width and height of the bounding box (in pixels) and class_id is the class id of the prediction.

For example, for a sample with 2 bounding boxes, the label format may be:
``tensor([[1, 8.4, 50.2, 100, 100], [5, 26.4, 10.1, 20, 40]])``.

Accepted Prediction Format
--------------------------
Deepchecks' checks use the :func:`~deepchecks.vision.DetectionData.infer_on_batch` function in order to get the predictions of the model in the correct format.
The accepted prediction format is a list of length N containing tensors of shape (B, 6), where N is the number
of images, B is the number of bounding boxes detected in the sample and each bounding box is represented by 6
values: ``[x_min, y_min, w, h, confidence, class_id]``.

>     x_min,y_min,w and h represent the bounding box location as before, confidence is the confidence score given by the model to

Reviewer: Not sure what I think, but consider writing explicitly and not "as before" - what if someone just wants to look at this and not at the label section?

Author: Think it is the same as in image format that refers to a different page for details.

As above will be better?

Reviewer: Yeah better IMO

            return imgs

        def batch_to_labels(self, batch):
            """Convert a batch bounding boxes to the required format.

            Parameters
            ----------

>             batch : torch.Tensor (N, B, 5)

Reviewer: Technically in this example the back contains both image and label.
Also, is the original batch label a tensor? Given that B is different for each image?

                The batch of bounding boxes to convert.

            Returns
            -------
            List
                A list of size N containing tensors of shape (B,5).
            """

            # each bbox in the labels is (class_id, x, y, x, y). convert to (class_id, x, y, w, h)
            bboxes = []

>             for image in batch[1]:

Reviewer: It's not an image it's label

                bboxes_in_image = [torch.cat((bbox[0], bbox[1:3], bbox[4:] - bbox[1:3]), dim=0) for bbox in image]
                if len(bboxes_in_image) != 0:
                    bboxes.append(torch.stack(bboxes_in_image))
            return bboxes

        def infer_on_batch(self, batch, model, device):
            """Get the predictions of the model on a batch of images.

            Parameters
            ----------
            batch : tuple

>                 The batch of data, containing images and bounding boxes.

Reviewer: can copy this line for previous function
